src/Geometry/coordinates.jl

```diff
@@ -67,13 +67,17 @@ Cartesian123Point(pt::Cartesian13Point{FT}) where {FT} =
     Cartesian123Point(pt.x1, zero(FT), pt.x3)
 
 @pointtype LatLongPoint lat long
+@pointtype LatLongZPoint lat long z
 
 product_coordinates(xp::XPoint, yp::YPoint) = XYPoint(promote(xp.x, yp.y)...)
 product_coordinates(xp::XPoint, zp::ZPoint) = XZPoint(promote(xp.x, zp.z)...)
 
 product_coordinates(xyp::XYPoint, zp::ZPoint) =
     XYZPoint(promote(xyp.x, xyp.y, zp.z)...)
 
+product_coordinates(latlongp::LatLongPoint, zp::ZPoint) =
+    LatLongZPoint(promote(latlongp.lat, latlongp.long, zp.z)...)
```

Comment:
What does exactly a `LatLongZPoint` represent? Like a point in spherical coordinates, where the third dimension is along the radial direction? Should it be called `LatLongRPoint` then?

Reply:
Probably, elevation measure from planet surface!

Reply:
I think z makes sense, as it represents height above the surface, but I'm fine either way. @simonbyrne What do you think?

Reply:
Yes, it's the distance from the surface of the sphere: we could have used radius (i.e. distance from the origin, but that would be somewhat less convenient, and would mean that the box model has a different vertical coordinate. I tried to lay this out in #81.

Reply:
ok, but if one has in mind the usual picture for spherical coordinates (see for instance this one from Wiki) then I think the `z` would be confusing as it would mean something parallel to the global z-axis, which is not what this represents
![image](https://user-images.githubusercontent.com/39932030/146130222-48c8fbfe-7270-4738-b95c-81f1bafc7ab4.png)

Reply:
Shall we move the discussion to #81 ? I think it will affect many things, not only this PR.